### ISSUE TITLE
Interfere in pulseaudio volume between mpv and libmpv 

### ISSUE THREAD

**Issue by xylosper** [mpv-player/mpv#1578]
PulseAudio supports per-application volume and it kept by PulseAudio.

When I changed volume in mpv, the changed volume also applied in libmpv client.
I've tried to set `audio-client-name` but it only changes the string in volume control and didn't solve the problem.

Is there any solution for this?

**Comment by wm4** [comment]
The ``audio-client-name`` option should handle this. Not sure why it wouldn't work?

**Comment by xylosper** [comment]
No idea. When launched client only, the volume control displays the name what I set to `audio-client-name` but if mpv is running, libmpv client does not even appear in volume control. Defintely they shares same control.

**Comment by wm4** [comment]
I think this happens when the audio-device-list property is queried. It creates a connection without the proper client name set. It should disconnect immediately, though, and I see no reason why it would override the stream name. (Or maybe this is a different bug.)

**Comment by xylosper** [comment]
Partially fixed. Now I can see both controls in mixer window and I can control them independently when they're running at the same time.

However, for instacne, If I execute client after excuted `mpv --mute <video>`, then the initial volume of client is muted.

Now I doubt pulseaudio because they're synced even if I can see separated controls.

**Comment by wm4** [comment]
Looks like PulseAudio loves to force application devs to add shitty singletons.

**Comment by wm4** [comment]
Or maybe I'm misunderstanding... needs more investigation.

**Comment by wm4** [comment]
(Yeha, I was misunderstanding.)

I tested it, I can't see that they conflict in any way. What might cause confusion is that the stream title contains "mpv" by default. You can override it by setting the ``title`` option.

**Comment by xylosper** [comment]
You can't reproduce this?

Setting `title` doesn't work. In fact, I've set `title` to ""(empty string) always.

**Comment by wm4** [comment]
>You can't reproduce this?

No.

>Setting title doesn't work. In fact, I've set title to ""(empty string) always.

Then maybe dump and check the values ao_pulse.c passes to the pulseaudio APIs.

**Comment by xylosper** [comment]
Finally I found the source. It's the line set PA_PROP_MEDIA_ROLE:
```
(void)pa_proplist_sets(proplist, PA_PROP_MEDIA_ROLE, "video");
```
If I comment out that line, no more interfere exists.

I don't know the exact purpose of each role because the document says nothing, it seems that the PA_PROP_MEDIA_ROLE is designed for making it possible to interfere between streams using PulseAudio at the very first time.
For instance, if a 'phone' role  stream is started from any application, all other streams are muted including external applications.

I found that VLC also sets "video" role and I can see that VLC and mpv also interferes each other.

With some test, "video" and "music" also interferes but "game" does not interfere.
Also, no specific role(which is set "none" by internal PulseAudio) does not cause interference.
